packages/contact-exchange/src/services/DIDCommService.ts

```diff
@@ -0,0 +1,156 @@
+import { Message, SecretsResolver, Secret } from 'didcomm-node';
+import { v4 as uuidv4 } from 'uuid';
+import { MessageTyp, MessageType } from './DIDCommOOBInvitation';
+import { DidPeerMethod } from '@adorsys-gis/multiple-did-identities/src/did-methods/DidPeerMethod';
+import { PeerDIDResolver } from 'did-resolver-lib';
+
+class DidcommSecretsResolver implements SecretsResolver {
+  private knownSecrets: Secret[];
+
+  constructor(knownSecrets: Secret[]) {
+    this.knownSecrets = knownSecrets;
+  }
+
+  async get_secret(secretId: string): Promise<Secret | null> {
+    return this.knownSecrets.find((secret) => secret.id === secretId) || null;
+  }
+
+  async find_secrets(secretIds: string[]): Promise<string[]> {
+    return secretIds.filter((id) =>
+      this.knownSecrets.some((secret) => secret.id === id),
+    );
+  }
+}
+
+function createSecretsResolver(knownSecrets: Secret[]): DidcommSecretsResolver {
+  return new DidcommSecretsResolver(knownSecrets);
+}
+
+export interface PrivateKeyJWK {
+  id: string;
+  type: string;
+  privateKeyJwk: {
+    crv: string;
+    d: string;
+    kty: string;
+    x: string;
+    y?: string;
+  };
+}
+
+function prependDidToSecretIds(
+  secrets: PrivateKeyJWK[],
+  did: string,
+): PrivateKeyJWK[] {
+  return secrets.map((secret) => ({
+    ...secret,
+    id: `${did}${secret.id.replace(did, '')}`,
+  }));
+}
+
+export async function processMediatorOOB(oob: string) {
+  try {
+    const oobParts = oob.split('=');
+    if (oobParts.length < 2) {
+      throw new Error('Invalid OOB format. Missing encoded payload.');
+    }
+
+    const oobUrl = oobParts[1];
+    const decodedOob = JSON.parse(
+      Buffer.from(oobUrl, 'base64url').toString('utf-8'),
+    );
+
+    if (!decodedOob.from) {
+      throw new Error('Invalid OOB content. Missing "from" field.');
+    }
```

Comment:
Don't we already have a similar logic somewhere?

Reply:
Yes we do have a similar logic in the OOBParser file, i reimplemented it to avoid some bugs.